client/signup/utils.js

```diff
@@ -17,16 +17,18 @@ var i18nUtils = require( 'lib/i18n-utils' ),
 	formState = require( 'lib/form-state' );
 
 function getFlowName( parameters ) {
-	var currentFlowName = flows.currentFlowName;
-	if ( parameters.flowName && isFlowName( parameters.flowName ) ) {
-		return parameters.flowName;
-	}
+	const flow = ( parameters.flowName && isFlowName( parameters.flowName ) ) ? parameters.flowName : defaultFlowName;
```

Comment:
`parameters` is an object, so I don't believe the `parameters.flowName &&` is necessary.

I think this might be a little clearer as one function where we assume `filterFlowName` is present and a function that doesn't return bad flow names.

```js
function getFlowName( parameters ) {
  const flow = isFlowName( parameters.flowName ) ? parameters.flowName : defaultFlowName;

  return flows.filterFlowName( flow );
}
```

If `filterFlowName` is a string, it is ignored, right? I think we should just assume a) that if it is present, it is a function and maybe b) that it is always present.